compiler/src/main/java/com/squareup/anvil/compiler/codegen/PsiUtils.kt

```diff
@@ -239,18 +239,31 @@ internal fun PsiElement.requireFqName(
   // First look in the imports for the reference name. If the class is imported, then we know the
   // fully qualified name.
   importPaths
-      .filter { it.alias == null }
-      .firstOrNull {
-        it.fqName.shortName().asString() == classReference
+      .filter { it.alias == null && it.fqName.shortName().asString() == classReference }
+      .also { matchingImportPaths ->
+        when {
+          matchingImportPaths.size == 1 ->
+            return matchingImportPaths[0].fqName
+          matchingImportPaths.size > 1 ->
+            return matchingImportPaths.first { importPath ->
```

Comment:
nit: would `.single` instead of `.first` make sense for the extra assertion that there should be one *and only one* correct import? (same question for line 261 below)

Reply:
While true resolving types is expensive. In Kotlin you can't import classes with the same name from different packages (neither in Java). So we can assume that there is only one.